lib/rspec/mocks/argument_matchers.rb

```diff
@@ -116,7 +116,7 @@ def kind_of(klass)
 
       # @private
       def self.anythingize_lonely_keys(*args)
-        hash = args.last.class == Hash ? args.delete_at(-1) : {}
+        hash = args.last.is_a?(Hash) ? args.delete_at(-1) : {}
```

Comment:
Please use `Hash === args.last`.  The reason being: RSpec winds up being used with all manner of user objects, including ones that do not define `is_a?` properly.  (One such example is `BasicObject`--`BasicObject.is_a?(Hash)` raises an error).  We trust that `Hash.===` is defined properly since it's a method on an object defined and owned by Ruby.